obelisk/cpp/obelisk_cpp/include/obelisk_node.h

```diff
@@ -0,0 +1,849 @@
+            rclcpp_lifecycle::LifecycleNode::on_shutdown(prev_state);
+
+            // Clear current data
+            callback_groups_.clear();
+
+            // Clean up publishers
+            for (auto& [key, pub] : publishers_) {
+                if (pub) {
+                    pub->Release();
+                }
+            }
+            publishers_.clear();
+            registered_publishers_
+                .clear(); // TODO: Consider if this should be called, or if I should leave this in place
+
+            // Clean up subscriptions
+            for (auto& [key, sub] : subscriptions_) {
+                if (sub) {
+                    sub->Release();
+                }
+            }
+            subscriptions_.clear();
+            registered_subscriptions_
+                .clear(); // TODO: Consider if this should be called, or if I should leave this in place
+
+            // Clean up timers
+            for (auto& [key, timer] : timers_) {
+                if (timer) {
+                    timer->cancel(); // Stop the timer
+                    timer.reset();   // Release the timer
+                }
+            }
+            timers_.clear();
+            registered_timers_.clear();
+
+            RCLCPP_INFO_STREAM(this->get_logger(), this->get_name() << " shutdown.");
+            return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+        }
+
+        rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn virtual PostConfigure(
+            __attribute__((unused)) const rclcpp_lifecycle::State& prev_state) {
+            return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+        }
+
+        rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn virtual PostActivate(
+            __attribute__((unused)) const rclcpp_lifecycle::State& prev_state) {
+            return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+        }
+
+        rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn virtual PostDeactivate(
+            __attribute__((unused)) const rclcpp_lifecycle::State& prev_state) {
+            return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+        }
+
+        rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn virtual PostCleanup(
+            __attribute__((unused)) const rclcpp_lifecycle::State& prev_state) {
+            return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+        }
+
+        rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn virtual PostShutdown(
+            __attribute__((unused)) const rclcpp_lifecycle::State& prev_state) {
+            return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+        }
```

Comment:
refer to slack discussion about this - this comment is a placeholder for that pre-merge